### PR TITLE
Tj/shopify logic

### PR DESCRIPTION
**Pull request by littlejohntj**
Moved around a bunch of shopify logic. Originally this change was to move logic from backend into a package. This ended up causing a problem with next.js and crypto-js which I couldn't figure out so I moved the logic into payment-ui where the backend will now be hosted.

### CODE REVIEW

File: apps/payment-ui/src/models/install-query-params.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { object, string, InferType } from 'yup'
+
+export let appInstallQueryParmSchema = object().shape({
+    hmac: string().optional(), // optional for the purpose of validation. i should do an extra check here or make a new structure
+    shop: string()
+        .matches(/^https?\:\/\/[a-zA-Z0-9][a-zA-Z0-9\-]*\.myshopify\.com\/?/)
```
Comment (askibin) [inline apps/payment-ui/src/models/install-query-params.model.ts:6]: where this regex comes from? what about non-latin domains, sub-domains, www. , are these not possible?

File: packages/transaction-builder/dist/index.js [hunk 2]
```diff
@@ -6087,119 +6087,186 @@ var require_browser = __commonJS({
...
+      const signMightNotMatch = x % y2;
+      if (sign2(y2) !== sign2(signMightNotMatch)) {
+        return signMightNotMatch + y2;
+      }
+      return signMightNotMatch;
+    }
+    function censorNegativeZero(x) {
+      return x === 0 ? 0 : x;
+    }
+    function createIntegerConversion(bitLength, typeOpts) {
```
Comment (askibin) [inline packages/transaction-builder/dist/index.js:6140]: what all this math is for?

File: apps/payment-ui/src/models/install-query-params.model.ts (hunk 1 above)
Comment (littlejohntj) [reply apps/payment-ui/src/models/install-query-params.model.ts:6]: check the bottom of step 4 

https://shopify.dev/docs/apps/auth/oauth/getting-started#step-4-confirm-installation

File: packages/transaction-builder/dist/index.js (hunk 2 above)
Comment (littlejohntj) [reply packages/transaction-builder/dist/index.js:6140]: that's just compiled typescript but good call out, I should add the dist/* to the .gitignore

File: apps/payment-ui/src/pages/api/install.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    const secret = process.env.SHOPIFY_SECRET_KEY
+
+    // Check for a secret key to decode with
+    if (secret == undefined) {
+        throw new Error('Did not have the required info to verify.')
+    }
+
+    const digest = crypto.HmacSHA256(queryStringAfterRemoving, secret)
+    const digestString = digest.toString()
```
Comment (harshasomisetty) [inline apps/payment-ui/src/pages/api/install.ts:89]: minor but simplify to

`const digestString = crypto.HmacSHA256(queryStringAfterRemoving, secret).toString()`

**Pull request closed without merging**